File: sources/academy/webscraping/scraping_basics_python/exercises/wikipedia_countries.py
import httpx
from bs4 import BeautifulSoup

url = "https://en.wikipedia.org/wiki/List_of_sovereign_states_and_dependent_territories_in_Africa"
response = httpx.get(url)
response.raise_for_status()

soup = BeautifulSoup(response.text, "html.parser")

for table in soup.select(".wikitable"):
    for row in table.select("tr"):
        cells = row.select("td")
        if cells:
            third_column = cells[2]
            link = third_column.select_one("a")
            if link:
                print(link.text)

File: sources/academy/webscraping/scraping_basics_javascript/exercises/wikipedia_countries.mjs
import * as cheerio from 'cheerio';

const url = 'https://en.wikipedia.org/wiki/List_of_sovereign_states_and_dependent_territories_in_Africa';
const response = await fetch(url);

if (!response.ok) {
  throw new Error(`HTTP ${response.status}`);
}

const html = await response.text();
const $ = cheerio.load(html);

for (const tableElement of $('.wikitable').toArray()) {
  const $table = $(tableElement);
  const rows = $table.find('tr');

  for (const rowElement of rows.toArray()) {
    const $row = $(rowElement);
    const cells = $row.find('td');

    if (cells.length > 0) {
      const $thirdColumn = $(cells[2]);
      const $link = $thirdColumn.find('a').first();
      if ($link.length) {
        console.log($link.text());
      }
    }
  }
}
